analyzers/tests/SonarAnalyzer.UnitTest/TestCases/OperatorOverloadsShouldHaveNamedAlternatives.CSharp11.cs

```diff
@@ -0,0 +1,7 @@
+﻿using System.Threading.Tasks;
+using System;
+
+internal interface IOperator<T> where T : IOperator<T>
+{
+    static virtual T operator ++(T input) => input; // Noncompliant {{Implement alternative method 'Increment' for the operator '++'.}}
```

Comment:
Add a test case where the `++` operator comes from a default interface implementation and the minus operator is defined in a class implementing that interface (as in OperatorsShouldBeOverloadedConsistently.CSharp11.cs).

Reply:
This is not the right way to handle virtual operators. `Foo` does not get the implementation from `IOperator<T>`here. You need to derive interfaces instead. Replace the test case with
```cs
public interface IPlusOperator<T> where T: IPlusOperator<T>
{
    static virtual T operator ++(T input) => input; // Noncompliant
}
public interface IPlusMinusOperator<T>: IPlusOperator<T> where T: IPlusMinusOperator<T>
{
    static virtual T operator --(T input) => input; // Compliant ++ is enforced by IPlusOperator<T>
}
```

Reply:
@martin-strecker-sonarsource  I do not understand this example.
Also, the one I added makes not so much sense; re-seeing it now.

This rule checks if there's an operator overloaded and proposes to change its name (see S4096).

IMO only this example
```csharp
internal interface IOperator<T> where T : IOperator<T>
{
    static virtual T operator ++(T input) => input; // Noncompliant {{Implement alternative method 'Increment' for the operator '++'.}}
}
```
To check that overloaded operators inside interfaces are detected.

Am I missing something else?

Reply:
Ah I see. It is really confusing having so many rules about almost the same topics in the same PR. It would have been better to create a PR per case.
I think, if you just add this three forms it would be ok:
```cs
public interface IOperators<T> where T: IOperators<T>
{
    static virtual T operator ++(T input) => input;
    static abstract T operator --(T input);
    static IOperators<T> operator ~(IOperators<T> input) => input;
}
```

Reply:
Indeed - it was not a good choice in hindsight - next time I'll go with one PR per rule.
Sorry for the confusion here.